src/lib/UI/LabelMaker/TranlationBasedLabelMaker.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\UI\LabelMaker;
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Makes human readable labels using translations.
+ * The item and label identifier are prefixed with a configured string, and translated in a configured domain.
+ *
+ * Example: $labelMaker->getLabel($myCustomTag, 'description').
+ *
+ * If no translation exists, uses the
+ */
+class TranlationBasedLabelMaker implements LabelMaker
+{
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /** @var string[] */
+    private $patterns;
+
+    /** @var string */
+    private $domain;
+
+    public function __construct(TranslatorInterface $translator, array $config = [])
+    {
+        $this->translator = $translator;
+        $this->patterns = $config['patterns'];
+        $this->domain = $config['domain'];
+    }
+
+    public function getLabel(string $identifier, $items, bool $generateDefault = true): string
+    {
+        if (!is_array($items)) {
```

Comment:
Unless you expect here a collection of objects (but I guess object with `__toString` could be the case), simple cast to array would do the trick as well.